synapse/util/caches/descriptors.py

```diff
@@ -335,20 +346,10 @@ def wrapped(*args, **kwargs):
             try:
                 cached_result_d = cache.get(cache_key, callback=invalidate_callback)
 
-                observer = cached_result_d.observe()
-                if DEBUG_CACHES:
```

Comment:
aw, I liked the read-through cache it was useful for debugging.
If you really want to remove it from here you should probably remove the `DEBUG_CACHES` variable entirely.